Multisig/UI/App/UIKit/TransactionlListViewController/TransactionListViewController.swift

```diff
@@ -0,0 +1,206 @@
+                            (error as NSError).domain == NSURLErrorDomain {
+                            return
+                        }
+                        self.onError(error)
+                    }
+                case .success(let page):
+                    var model = TransactionsListViewModel(page.results.flatMap { TransactionViewModel.create(from: $0) })
+                    model.next = page.next
+
+                    DispatchQueue.main.async { [weak self] in
+                        guard let `self` = self else { return }
+                        self.model.append(from: model)
+                        self.onSuccess()
+                    }
+                }
+
+                DispatchQueue.main.async { [weak self] in
+                    self?.stopNextPageLoadingAnimation()
+                }
+
+                self.loadNextPageDataTask = nil
+            }
+
+        } catch {
+            onError(error)
+        }
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        model.sections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        model.sections[section].transactions.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueCell(TransactionListTableViewCell.self, for: indexPath)
+
+        // remove existing controller
+        if let vc = cell.controller {
+            vc.willMove(toParent: nil)
+            vc.view.removeFromSuperview()
+            vc.removeFromParent()
+        }
+
+        // add new controller
+        let tx = model.sections[indexPath.section].transactions[indexPath.row]
+        let vc = UIHostingController(rootView: TransactionCellView(transaction: tx))
+        addChild(vc)
+        cell.setController(vc)
+        vc.didMove(toParent: self)
+
+        // somehow adding hosting controller auto-shows the navigation bar
+        navigationController?.navigationBar.isHidden = true
+
+        // if last cell, trigger the loading operation
+        if indexPath.section == model.sections.count - 1 &&
+            indexPath.row == model.sections[indexPath.section].transactions.count - 1 {
+            loadNextPage()
+        }
+        return cell
+    }
+
+    // for getting cell size
+    private var cellSizeController: UIHostingController<TransactionCellView>!
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let tx = model.sections[indexPath.section].transactions[indexPath.row]
+
+        if cellSizeController == nil {
+            cellSizeController = UIHostingController<TransactionCellView>(rootView: TransactionCellView(transaction: tx))
+        } else {
+            cellSizeController.rootView = TransactionCellView(transaction: tx)
+        }
+        let size = cellSizeController.sizeThatFits(in: tableView.bounds.size)
```

Comment:
I thought it should always be the same, but this approach is more flexible 👍

Reply:
the cells are of different size